phabricatoremails/model/phabricator_secure.py

```diff
@@ -0,0 +1,115 @@
+from dataclasses import dataclass
```

Comment:
There are a few commonalities between this module and `model.phabricator`. Many of these classes could extend their non-secure counterparts, and thus making it more obvious what is missing or what is added to which classes.

Another approach to reducing repetition here could be to implement a single class for both secure and non-secure revisions, and pass a secure flag in order to remove (or add) a particular field.

Reply:
I agree that that would remove lines of code, but I think that that would add significant complexity. There's a value in being able to look up an event and immediately see what properties it has without having to parse a flagging structure.

Note that neither type of event is consistently a subset of the other. For example, secure events with comments have `commentCount`, while public events with comments have `mainComment` and `inlineComments`. You _could_ make these `Optional`, but this sacrifices readability (when is this property available?) so that there's less lines of code. I don't think that this tradeoff is worth it.